corehq/apps/integration/views.py

```diff
@@ -126,18 +137,22 @@ def get_post_data_for_otp(request, domain):
     try:
         case_id = request.POST['case_id']
         post_params = {
-            'mobile': request.POST.get("phone_number", None),
-            'testDate': request.POST['test_date'],
-            'onsetDate': request.POST['onset_date'],
+            'jobId': str(uuid4()),
         }
     except KeyError as ke:
         raise InvalidOtpRequestException(_("OTP Request missing a required argument %s") % ke.args[0])
 
     case = CaseAccessors(domain).get_case(case_id)
-    properties = case.dynamic_case_properties()
 
-    if 'contact_phone_number' in properties:
-        post_params['mobile'] = properties['contact_phone_number']
+    property_map = {
+                    'phone_number': 'mobile',
+                    'test_date': 'testDate',
+                    'onset_date': 'onsetDate',
+                    'test_type': 'testType',
+                   }
+    for request_param in property_map:
+        if request_param in request.POST:
+            post_params[property_map[request_param]] = request.POST[request_param]
```

Comment:
nit: python has `items` to loop through values in a dict:

```suggestion
    for request_param, post_param in property_map.items():
        if request_param in request.POST:
            post_params[post_param] = request.POST[request_param]
```